Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/CloudFormationWriter.cs

```diff
@@ -283,8 +284,18 @@ private void RemoveOrphanedLambdaFunctions(HashSet<string> processedLambdaFuncti
                 return;
             }
 
+            var resources = new List<string>();
+            if (_templateWriter is YamlWriter)
```

Comment:
Putting this logic in `CloudFormationWriter` breaks the abstraction of this class which is supposed to be agnostic of JSON or Yaml. Can you push this logic in `YamlWriter`'s `GetToken` method so that if the `T` is of type `Dictionary<string, object>` handle the conversion of `Dictionary<string, YamlMappingNode>` to `Dictionary<string, object>`?

Another option is to add to `ITemplateWriter` a `GetKeys(string path)` since all we need here is keys.

Reply:
Added a new function `GetKeys(string path)`